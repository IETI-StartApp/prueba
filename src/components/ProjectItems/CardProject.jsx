import React from 'react';
import {Avatar, Box, Card, CardContent, CardHeader, CardMedia, Typography} from '@material-ui/core';
import {useStyles} from './styles';
import ProgressGraph from './ProgressGraph';
import MonetizationOnIcon from '@material-ui/icons/MonetizationOn';
import SupervisedUserCircleIcon from '@material-ui/icons/SupervisedUserCircle';

export default function CardProject({projectId}) {

    const classes = useStyles();
// pass it as props
    const ownerImage = "https://www.dzoom.org.es/wp-content/uploads/2020/02/portada-foto-perfil-redes-sociales-consejos.jpg";
    const owner = "Nombre de usuario";
    const creationDate = new Date(2021, 10, 2);
    const projectImage = "https://e00-expansion.uecdn.es/assets/multimedia/imagenes/2018/07/25/15325097324389.jpg";
    const pReturn = 15;
    const fObjective = 30000000;
    const projectName = "Nombre del proyecto";
    const financed = 21000000;
    const partners = 10;

    return (
        <Card className={classes.sizeCard}>
            <CardHeader
                avatar={
                    <Avatar aria-label="photo" src={ownerImage}>
                    </Avatar>
                }
                title={owner}
                subheader={creationDate.toLocaleDateString()}
            />
            <CardMedia
                component="img"
                alt="Project image"
                height="210"
                image={projectImage}
            />
            <CardContent>
                <Typography gutterBottom variant="h5" component="h2">
                    {projectName}
                </Typography>
                <Typography variant="body2" component="p">
                    {pReturn}% de retorno efectivo anual
                </Typography>
                <Typography variant="body2" component="p">
                    $ {fObjective} Meta de financiaci√≥n
                </Typography>
                <Box position="relative" display="inline-flex" mt={2}>
                    <Box mr={5}>
                        <Box position="relative" display="inline-flex">
                            <Avatar className={classes.avatar}><MonetizationOnIcon/></Avatar>
                        </Box>
                        <Typography> Financiado </Typography>
                        <Typography className={classes.mainColor}> $ {financed}</Typography>
                    </Box>
                    <Box mr={5}>
                        <ProgressGraph/>
                        <Typography> Progreso </Typography>
                    </Box>
                    <Box>
                        <Box position="relative" display="inline-flex">
                            <Avatar className={classes.avatar}><SupervisedUserCircleIcon/></Avatar>
                        </Box>
                        <Typography> Socios </Typography>
                        <Typography className={classes.mainColor}>{partners}</Typography>
                    </Box>
                </Box>
            </CardContent>
        </Card>
    );
}
