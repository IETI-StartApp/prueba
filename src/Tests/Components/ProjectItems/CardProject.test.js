import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import CardProject from "../../../../src/components/ProjectItems/CardProject";
import { Typography, CardHeader, CardMedia } from '@material-ui/core';

describe('Testing card project', () => {

    const wrapper = shallow(<CardProject/>)

    test('should load CardProject', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test ('should load a valid Card Header', () => {
        const cardHeader = wrapper.find(CardHeader).at(0);

        expect(cardHeader.props().title).toBeDefined();
        expect(cardHeader.props().subheader).toBeDefined();
        expect(cardHeader.props().avatar).toBeDefined();

        expect(cardHeader.props().title).not.toEqual("");
        expect(cardHeader.props().subheader).not.toEqual("");
    })

    test ('should load the project image', () => {
        const image = wrapper.find(CardMedia).at(0).props().image;
        expect(image).toBeDefined();
        expect(image).not.toEqual("");
    })

    test ('should load a valids Typographys', () => {
        const projectName = wrapper.find(Typography).at(0).text();
        expect(projectName).toBeDefined();
        expect(projectName).not.toEqual("");

        const returnPercentage  = wrapper.find(Typography).at(1).text().split("%")[0];
        expect(returnPercentage).not.toBeNaN();
        expect(parseFloat(returnPercentage)).toBeGreaterThanOrEqual(0);
        expect(parseFloat(returnPercentage)).toBeLessThanOrEqual(100);
        
        const fObjective  = wrapper.find(Typography).at(2).text().split(" Meta")[0];
        expect(fObjective).not.toBeNaN();

        const financed  = wrapper.find(Typography).at(4).text();
        expect(financed.substring(4, financed.length - 1)).not.toBeNaN();
        expect(parseFloat(financed.substring(4, financed.length - 1))).toBeGreaterThan(0);

        const partners  = wrapper.find(Typography).at(7).text();
        expect(partners).not.toBeNaN();
        expect(parseFloat(partners)).toBeGreaterThanOrEqual(0);
    })


})