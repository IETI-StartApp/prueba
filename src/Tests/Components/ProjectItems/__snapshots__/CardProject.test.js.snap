// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing card project should load CardProject 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-sizeCard-1"
>
  <WithStyles(ForwardRef(CardHeader))
    avatar={
      <WithStyles(ForwardRef(Avatar))
        aria-label="photo"
        src="https://www.dzoom.org.es/wp-content/uploads/2020/02/portada-foto-perfil-redes-sociales-consejos.jpg"
      />
    }
    subheader="11/2/2021"
    title="Nombre de usuario"
  />
  <WithStyles(ForwardRef(CardMedia))
    alt="Project image"
    component="img"
    height="210"
    image="https://e00-expansion.uecdn.es/assets/multimedia/imagenes/2018/07/25/15325097324389.jpg"
  />
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      component="h2"
      gutterBottom={true}
      variant="h5"
    >
      Nombre del proyecto
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      component="p"
      variant="body2"
    >
      15
      % de retorno efectivo anual
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      component="p"
      variant="body2"
    >
      $ 
      30000000
       Meta de financiaci√≥n
    </WithStyles(ForwardRef(Typography))>
    <Styled(MuiBox)
      display="inline-flex"
      mt={2}
      position="relative"
    >
      <Styled(MuiBox)
        mr={5}
      >
        <Styled(MuiBox)
          display="inline-flex"
          position="relative"
        >
          <WithStyles(ForwardRef(Avatar))
            className="makeStyles-avatar-3"
          >
            <Memo([object Object]) />
          </WithStyles(ForwardRef(Avatar))>
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Typography))>
           Financiado 
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-mainColor-2"
        >
           $ 
          21000000
        </WithStyles(ForwardRef(Typography))>
      </Styled(MuiBox)>
      <Styled(MuiBox)
        mr={5}
      >
        <ProgressGraph />
        <WithStyles(ForwardRef(Typography))>
           Progreso 
        </WithStyles(ForwardRef(Typography))>
      </Styled(MuiBox)>
      <Styled(MuiBox)>
        <Styled(MuiBox)
          display="inline-flex"
          position="relative"
        >
          <WithStyles(ForwardRef(Avatar))
            className="makeStyles-avatar-3"
          >
            <Memo([object Object]) />
          </WithStyles(ForwardRef(Avatar))>
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Typography))>
           Socios 
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-mainColor-2"
        >
          10
        </WithStyles(ForwardRef(Typography))>
      </Styled(MuiBox)>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
